app/models/manageiq/providers/openshift/infra_manager/refresh_worker/runner.rb

```diff
@@ -0,0 +1,168 @@
+require 'concurrent/atomic/atomic_boolean'
+
+class ManageIQ::Providers::Openshift::InfraManager::RefreshWorker::Runner < ManageIQ::Providers::Kubevirt::InfraManager::RefreshWorker::Runner
+
+  private
+
+  #
+  # Returns the reference to the manager.
+  #
+  # @return [ManageIQ::Providers::Kubevirt::InfraManager] The manager.
+  #
+  def manager
+    @manager ||= ManageIQ::Providers::Openshift::InfraManager.find(@cfg[:ems_id])
```

Comment:
I'm surprised so much of this overriding is necessary.  Makes me wonder if there's a better way, such as providing a method like `manageiq_class`, which returns `ManageIQ::Providers::Openshift::InfraManager`, and then the actual logic would be something like

```
    @manager ||= manager_class.find(@cfg[:ems_id])
```

Reply:
This is the simple example of course - the same comment really applies to all of the methods in this file.